enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -300,57 +264,64 @@ The implementation changes needed for the proposed enhancement are documented in
 
 #### Cluster DNS Operator
 
-Cluster DNS Operator will deploy CoreDNS with the ``egressfirewall`` plugin enabled by adding it to the corefile. For the EgressFirewall rules to
+Cluster DNS Operator will deploy CoreDNS with the `egressfirewall` plugin enabled by adding it to the corefile. For the EgressFirewall rules to
 apply consistently, even for DNS names that are resolved by custom upstreams, it will be added to all server blocks in the corefile. As the
-plugin will watch and update the ``EgressFirewallDNSName`` CRs in the ``dns.openshift.io`` api-group, proper RBAC permissions will be needed
-to be added to the ``ClusterRole`` for CoreDNS.
+plugin will watch and update the `EgressFirewallDNSName` CRs in the `dns.openshift.io` api-group, proper RBAC permissions will be needed
+to be added to the `ClusterRole` for CoreDNS.
+
+The new `EgressFirewallDNSName` controller will be added to the Cluster DNS Operator. The controller will watch the `EgressFirewallDNSName` CRs,
```

Comment:
It would be better if the design only enabled this controller when there was an egress firewall configured.  If customers do not use that functionality, this adds complicated operations that are not required, and risks regressions with current DNS operations.  

Also, at a minimum, we need to see discussion of whether the status of this new controller will impact the status of the cluster-dns-operator overall.  Are there any conditions of the new controller which wouid make cluster-dns-operator degraded?

Reply:
> It would be better if the design only enabled this controller when there was an egress firewall configured. If customers do not use that functionality, this adds complicated operations that are not required, and risks regressions with current DNS operations.

The feature is currently planned for tech preview. So, the controller will be enabled if the specific feature gate is enabled. Even when it is enabled, the controller will not be receiving any events if egress firewall is not configured.



> Also, at a minimum, we need to see discussion of whether the status of this new controller will impact the status of the cluster-dns-operator overall. Are there any conditions of the new controller which wouid make cluster-dns-operator degraded?

The controller will be watching the new `EgressFirewallDNSName` CRs and will perform re-resolution based on the TTL of the DNS names which are added to the status of the CRs. This won't impact the status of cluster-dns-operator.